### PR TITLE
Feature/update gene

### PR DESCRIPTION
**Pull request by sibyl229**
**WARNING: Please ensure #26 is merged before reviewing this PR**.

Updating a gene should be working now. 

_An odd thing I noticed is that I had to restart the ring server after I switch branch_. Have you experienced similar problems?

### CODE REVIEW

File: client/src/containers/Gene/BaseForm.js [hunk 1]
```diff
@@ -84,14 +84,15 @@ class FormDataStore {
...
-          ...idSegments.reduceRight((result, idSegment, index) => {
-            return {
-              [idSegment]: result,
-            };
-          }, value),
-          ...result,
-        };
+        idSegments.reduce((resultSubtree, idSegment, index) => {
+          if (index < idSegments.length - 1) {
+            resultSubtree[idSegment] = resultSubtree[idSegment] || {};
```
Comment (mgrbyte) [inline client/src/containers/Gene/BaseForm.js:89]: maybe `resultSubtree[idSegment] |=  {}`

File: client/src/containers/Gene/GeneProfile.js [hunk 2]
```diff
@@ -144,55 +168,43 @@ class GeneProfile extends Component {
...
-                this.setState({
-                  data: data,
-                }, () => {
-                  this.closeSplitGeneDialog();
-                });
-              }}
-            /> :
-            null
-        }
+        <KillGeneDialog
```
Comment (mgrbyte) [inline client/src/containers/Gene/GeneProfile.js:177]: Could the name used be either $state.cgcName or $state.sequenceName?

**Review by mgrbyte — COMMENTED** [review]
Apart from the question of cgcName/seqName in the merge & split dialogs,
i think this good :+1:

File: client/src/containers/Gene/BaseForm.js (hunk 1 above)
Comment (sibyl229) [reply client/src/containers/Gene/BaseForm.js:89]: I think javascript may not have the `||=` or `|=` syntax. I get syntax error when I tried that.

File: client/src/containers/Gene/GeneProfile.js (hunk 2 above)
Comment (sibyl229) [reply client/src/containers/Gene/GeneProfile.js:177]: Good catch! Gonna fix this.

File: client/src/containers/Gene/BaseForm.js (hunk 1 above)
Comment (mgrbyte) [reply client/src/containers/Gene/BaseForm.js:89]: hur seemed like it did to me, but is a minor thing anyway (https://www.w3schools.com/js/js_assignment.asp)

Comment (mgrbyte) [reply client/src/containers/Gene/BaseForm.js:89]: hmm, yeah, only seems to work for numbers (logical bit-wise assignment)

File: client/src/containers/Gene/GeneProfile.js [hunk 3]
```diff
@@ -118,6 +118,12 @@ class GeneProfile extends Component {
     });
   }
 
+  getDisplayName = (data) => (
+    data['gene/cgc-name'] ||
+    data['gene/sequence-name'] ||
+    data['gene/id']
```
Comment (mgrbyte) [inline client/src/containers/Gene/GeneProfile.js:124]: Should this not have a `return` statement? Or perhaps this is a feature of React/`Arrow functions` that I'm unaware of...

**Pull request merged**